batch2/batch/worker.py

```diff
@@ -348,6 +339,8 @@ def __init__(self, worker, parameters, cpu_sem):
 
         self._run_task = asyncio.ensure_future(self.run(cpu_sem))
 
+        self.last_updated = None
```

Comment:
Was there something you wanted to use this variable for?

Reply:
The linter gives an error if a member is assigned in another method but not in __init__.  I guess there is a danger, if you call the methods out of order, that you get an attribute error unexpectedly.

Reply:
FWIW, I find defining all the fields in the `__init__` method helpful when reading unfamiliar files. Especially given the lack of any type information.